src/handy.c

```diff
@@ -643,3 +643,31 @@ char *strlwr(char *s)
 	for(;*tmp;++tmp) *tmp=tolower((unsigned char)*tmp);
 	return s;
 }
+
+char *strip_fqdn(char **fqdn_p)
+{
+	char *fqdn=*fqdn_p;
+	char *non_fqdn=NULL;
+	char *tmp=NULL;
+	tmp=strchr(fqdn,'.');
+	if(tmp)
+	{
+		size_t s = strlen(fqdn)-strlen(tmp)+1;
+		int r;
+		non_fqdn=(char *)malloc_w(s*sizeof(char), __func__);
```

Comment:
If this returns NULL, it should be a fatal error. I would make the function return an int, and have 0 for success and -1 for error.
And I would make the place that calls it in src/conffile.c check and return -1 itself, if an error happens.

Actually, better than that would be to just truncate the existing string, because you wouldn't need to malloc and error check at all. Or snprintf into it below.

The whole function becomes this:

void strip_fqdn(char **fqdn)
{
  char *tmp;
  if(!fqdn || !*fqdn)
    return;
  if((tmp=strchr(*fqdn, '.')))
   *tmp='\0';
}

And I would probably leave it in src/conffile.c, next to where it is used, rather than putting it in handy.c.